conn.go

```diff
@@ -69,7 +69,7 @@ func (conn *Conn) Serve() {
 		line, err := conn.controlReader.ReadString('\n')
 		if err != nil {
 			if err == io.EOF {
-				continue
+				break
```

Comment:
You should log if the error is different than `io.EOF` and let it `break` otherwise as it's the default. More DRY.